introduce ollama processor (#2702)
redpanda-data/connect: docs/modules/components/pages/processors/ollama_embeddings.adoc

= ollama_embeddings
:type: processor
:status: experimental
:categories: ["AI"]

////
     THIS FILE IS AUTOGENERATED!

     To make changes, edit the corresponding source file under:

     https://github.com/redpanda-data/connect/tree/main/internal/impl/<provider>.

     And:

     https://github.com/redpanda-data/connect/tree/main/cmd/tools/docs_gen/templates/plugin.adoc.tmpl
////

component_type_dropdown::[]

> Processor that uses the Ollama API to create vector embeddings.

Reviewer: ```suggestion
Sends text to your chosen Ollama large language model (LLM) and creates vector embeddings, using the Ollama API. Vector embeddings are long arrays of numbers that represent values or objects, in this case text. 
```
I've suggested a more detailed description using a similar format to the chat processor.

Introduced in version 4.32.0.

[tabs]
======
Common::
+
--

```yml

> # Common config fields, showing default values

Reviewer: ```suggestion
# Common configuration fields, showing default values
```
Brings into line with the style guide.

label: ""
ollama_embeddings:

>   server_address: "" # No default (optional)

Reviewer: Isn't the local server address the default server address?

  model: nomic-embed-text # No default (required)
  text: "" # No default (optional)
```

--
Advanced::
+
--

```yml

> # All config fields, showing default values

Reviewer: ```suggestion
# All configuration fields, showing default values
```
Brings into line with the style guide.

label: ""
ollama_embeddings:
  server_address: "" # No default (optional)
  ollama_directory: "" # No default (optional)
  model: nomic-embed-text # No default (required)
  text: "" # No default (optional)
```

--
======

> Processor that uses the Ollama API to create vector embeddings.

Reviewer: Do we need this line?

> See https://ollama.com/[https://ollama.com/^] for more information.

Reviewer: ```suggestion
For more information, see the https://ollama.com/[Ollama website^].
```
Just to bring into line with the style guide.

== Fields

=== `server_address`

> The address of a remote ollama server to use. If omitted a local ollama server will be started and ran.

Reviewer: ```suggestion
The address of the Ollama server to use. By default, a local Ollama server starts and runs unless you specify the address of a remote server.
```

*Type*: `string`

=== `ollama_directory`

> The directory to unpack the ollama binary into. Defaults to /tmp

Reviewer: ```suggestion
Unpack the Ollama binary to this directory. Default is `/tmp`.

```

*Type*: `string`

=== `model`

> The ollama model to use. A full list is available at ollama.com/models

Reviewer: ```suggestion
The name of the Ollama LLM to use. For a full list of models, see the https://ollama.com/models[Ollama website].
```

*Type*: `string`

```yml
# Examples

model: nomic-embed-text

model: mxbai-embed-large

model: snowflake-artic-embed

model: all-minilm
```

=== `text`

> The text to create embeddings for. By default the entire payload as a string is submitted.

Reviewer: The text you want to create vector embeddings for. By default, the processor submits the entire payload as a string.
